lib/fluent/plugin/out_logdna.rb

```diff
@@ -14,11 +14,16 @@ class LogDNAOutput < Fluent::BufferedOutput
     config_param :app, :string, default: nil
     config_param :file, :string, default: nil
     config_param :ingester_domain, :string, default: 'https://logs.logdna.com'
-    config_param :request_timeout_seconds, :integer, default: 30
+    config_param :request_timeout, :string, default: '30'
 
     def configure(conf)
       super
       @host = conf['hostname']
+
+      timeout_match = @request_timeout.match(/[0-9]*/)[0]
+      timeout = timeout_match == "" ? 30 : timeout_match.to_i
+      ms_match = @request_timeout.include? "ms"
+      @request_timeout = ms_match && timeout_match != "" ? timeout / 1000 : timeout
     end
```

Comment:
Finally, I think it would be great to not have the prefixes baked so directly in the code. As it is it's gonna be hard to add new prefixes if we want to, and the chance of bugs coming up is increased. Base on this and the above comments I have the following suggestion:

```
# make these two variables globals
timeout_prefix_map = { s: 1, ms: 1000 }
timeout_regex = Regexp.new("^([0-9]+)\s*(#{timeout_prefix_map.keys.join("|")})$")

# this section goes into this part of the code
num_component = 30.0
prefix_component = 's'

timeout_regex.match(@request_timeout) do |match|
  num_component = match[1].to_f
  prefix_component = match[2]
end

@request_timeout = num_component / timeout_prefix_map[prefix_component.to_sym]
```

Reply:
Yep, that's *much* better. Thanks!